ledger-service/http-json/src/main/scala/com/digitalasset/http/json/DomainJsonDecoder.scala

```diff
@@ -211,6 +227,19 @@ class DomainJsonDecoder(
         .map(_.toOption.flatten.toRightDisjunction(JsonError(cannotResolveTemplateId(id))))
     )
 
+  private def templateId_(
+      id: domain.ContractTypeId.Template.OptionalPkg,
+      jwt: Jwt,
+      ledgerId: LedgerApiDomain.LedgerId,
+  )(implicit
+      ec: ExecutionContext,
+      lc: LoggingContextOf[InstanceUUID],
+  ): ET[domain.ContractTypeId.Template.Resolved] =
+    eitherT(
+      resolveTemplateId(jwt, ledgerId)(id)
+        .map(_.toOption.flatten.toRightDisjunction(JsonError(cannotResolveTemplateId(id))))
+    )
+
```

Comment:
I plan to address this duplication along with #14844; this is perfectly sensible here given the current state of resolution.

Reply:
Like so. #15282@40d266073029d764fa341cd73c86968871dcc6b2